The job **`x86_64-gnu-distcheck`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/17206307109/job/48807677356) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/48807677356) [(plain)](https://github.com/rust-lang/rust/commit/55a0a59cb31ad94fbf71278d24bbdfbbef89815f/checks/48807677356/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
fmt error: Not running formatting checks; rustfmt.toml does not exist.
fmt error: This may happen in distributed tarballs.
tidy check
fatal: not a git repository (or any of the parent directories): .git
tidy check error: failed to retrieve base commit: command did not execute successfully: "git" "rev-parse" "HEAD^1"
expected success, got: exit status: 128

No base commit, skipping rustdoc_json check
Checking tidy rustdoc_json...
Skipping error code explanation removal check
Cannot figure out the SHA of the GCC submodule
some tidy checks failed
Bootstrap failed while executing `test --stage 2`
Command `/tmp/distcheck/distcheck-plain-src/build/x86_64-unknown-linux-gnu/stage1-tools-bin/rust-tidy /tmp/distcheck/distcheck-plain-src /tmp/distcheck/distcheck-plain-src/build/x86_64-unknown-linux-gnu/stage0/bin/cargo /tmp/distcheck/distcheck-plain-src/build 4 npm` failed with exit code 1
Created at: src/bootstrap/src/core/build_steps/tool.rs:1583:23
Executed at: src/bootstrap/src/core/build_steps/test.rs:1225:29

Command has failed. Rerun with -v to see more details.
Build completed unsuccessfully in 0:01:50
make: *** [Makefile:49: check] Error 1
Bootstrap failed while executing `test distcheck`
Command `make check [workdir=/tmp/distcheck/distcheck-plain-src]` failed with exit code 2
Created at: src/bootstrap/src/core/build_steps/test.rs:3147:9
Executed at: src/bootstrap/src/core/build_steps/test.rs:3150:14

Command has failed. Rerun with -v to see more details.
Build completed unsuccessfully in 0:10:08
  local time: Mon Aug 25 10:54:58 UTC 2025
  network time: Mon, 25 Aug 2025 10:54:59 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

--- a/src/bootstrap/src/core/build_steps/test.rs
+++ b/src/bootstrap/src/core/build_steps/test.rs
@@ -3129,7 +3129,7 @@ impl Step for Distcheck {
 
         let configure_args: Vec<String> = std::env::var("DISTCHECK_CONFIGURE_ARGS")
             .map(|args| args.split(" ").map(|s| s.to_string()).collect::<Vec<String>>())
-            .unwrap_or(vec![]);
+            .unwrap_or_default();
 
         command("tar")
             .arg("-xf")
@@ -3146,6 +3146,9 @@ impl Step for Distcheck {
             .run(builder);
         command(helpers::make(&builder.config.host_target.triple))
             .arg("check")
+            // Do not run the build as if we were in CI, otherwise git would be assumed to be
+            // present, but we build from a tarball here
+            .env("GITHUB_ACTIONS", "0")
             .current_dir(&plain_src_dir)
             .run(builder);
 

